src/ocrd/processor/base.py

```diff
@@ -289,6 +295,17 @@ def list_resources(self):
             print(res)
         return
 
+    def _setup(self) -> None:
+        """
+        Validate parameters, then run :py:meth:`setup`. Called whenever
+        :py:data:`parameter` changes.
```

Comment:
```suggestion
        :py:data:`parameter` is re-assigned.
```

Because "changes" implies that `_setup` is called when doing e.g. `self.parameter['foo'] = 'bar'` when it is only triggered by the setter, e.g. `self.parameter = {'foo': 'bar'}`.

Reply:
Oh, indeed, that's very significant!

So perhaps in addition to that we should indeed use [collections.Mapping](https://docs.python.org/2/library/collections.html#collections.Mapping) instead of dict here, as you proposed?